portals/devportal/src/main/webapp/services/login/login_callback.jsp

```diff
@@ -84,8 +84,20 @@
         postLogoutRedirectURI = serverUrl + appContext + LOGOUT_CALLBACK_URL_SUFFIX;
     }
     String error = request.getParameter("error");
-    if (error != null && error.equals("login_required")) {
-        response.sendRedirect(postLogoutRedirectURI + "?referrer=" + referrer);
+    boolean isAnonymousEnabled = false;
+    if ((error != null) && error.equals("login_required")) {
+        if(state!=null){
+            state = URLDecoder.decode(state, "UTF-8");
+            if(state.contains("/apis") == true || state.contains("/home") == true){
```

Comment:
Could be simply rewritten as follows.

```javascript
if (state.contains("/apis") || state.contains("/home")) {
```

Reply:
Fixed in https://github.com/wso2/apim-apps/pull/523